src/mscorlib/src/System/ThrowHelper.cs

```diff
@@ -227,6 +248,94 @@ internal static class ThrowHelper {
                     argumentName = "sourceBytesToCopy";
                     break;
 
+                case ExceptionArgument.keys:
+                    argumentName = "keys";
+                    break;
+
+                case ExceptionArgument.elementType:
+                    argumentName = "elementType";
+                    break;
+
+                case ExceptionArgument.newSize:
+                    argumentName = "newSize";
+                    break;
+
+                case ExceptionArgument.len:
```

Comment:
Looks like it should work; Enum.ToString() looks like it could do with some improves - (though that's a different PR)